pkg/cmd/api/api.go

```diff
@@ -337,6 +348,17 @@ func fillPlaceholders(value string, opts *ApiOptions) (string, error) {
 	return filled, nil
 }
 
+func hostnameValidator(v interface{}) error {
```

Comment:
Hey Brandon, I was looking at the issue and saw your PR, so checking it out. Nice work!

What's the point of `hostnameValidator` taking an `interface{}` instead of a string? Won't the hostname always be a string?

If we decide to stick with taking an `interface{}`, we should do a [safe type assertion](https://tour.golang.org/methods/15) to avoid a panic.

Reply:
@cksriprajittichai I completely agree with your suggestion. Once I get some input on the best place to put this function for it to be shared, I'll be sure to implement your change! I can't see a use case where this would take something other than a string.

Reply:
Hey @wilso199 
Maybe `utils` folder would be a good place for this function. Then you can use it here and also update `login.go` file.